neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -0,0 +1,298 @@
+                p.Count,
+                Weight = w
+            }).WeightedAverage(p => p.Count, p => p.Weight);
+            count = Math.Max(count, Blockchain.StandbyValidators.Length);
+            HashSet<ECPoint> sv = new HashSet<ECPoint>(Blockchain.StandbyValidators);
+            return engine.Service.Snapshot.Storages.Find(new[] { Prefix_Validator }).Select(p => new
+            {
+                PublicKey = p.Key.Key.Skip(1).ToArray().AsSerializable<ECPoint>(),
+                ValidatorState.FromByteArray(p.Value.Value).Votes
+            }).Where(p => (p.Votes.Sign > 0) || sv.Contains(p.PublicKey)).OrderByDescending(p => p.Votes).ThenBy(p => p.PublicKey).Select(p => p.PublicKey).Take(count).OrderBy(p => p).ToArray();
+        }
+
+        public class AccountState : Nep5AccountState
+        {
+            public uint BalanceHeight;
+            public ECPoint[] Votes;
+
+            public AccountState()
+            {
+                this.Votes = new ECPoint[0];
+            }
+
+            public AccountState(byte[] data)
+                : base(data)
+            {
+            }
+
+            protected override void FromStruct(Struct @struct)
+            {
+                base.FromStruct(@struct);
+                BalanceHeight = (uint)@struct[1].GetBigInteger();
+                Votes = @struct[2].GetByteArray().AsSerializableArray<ECPoint>(Blockchain.MaxValidators);
+            }
+
+            protected override Struct ToStruct()
+            {
+                Struct @struct = base.ToStruct();
+                @struct.Add(BalanceHeight);
+                @struct.Add(Votes.ToByteArray());
+                return @struct;
+            }
+        }
+
+        internal class ValidatorState
+        {
+            public BigInteger Votes;
+
+            public static ValidatorState FromByteArray(byte[] data)
+            {
+                return new ValidatorState
+                {
+                    Votes = new BigInteger(data)
+                };
+            }
+
+            public byte[] ToByteArray()
+            {
+                return Votes.ToByteArray();
+            }
+        }
+
+        internal class ValidatorsCountState
+        {
+            public BigInteger[] Votes = new BigInteger[Blockchain.MaxValidators];
+
+            public static ValidatorsCountState FromByteArray(byte[] data)
+            {
+                using (MemoryStream ms = new MemoryStream(data, false))
+                using (BinaryReader r = new BinaryReader(ms))
+                {
+                    BigInteger[] votes = new BigInteger[(int)r.ReadVarInt(Blockchain.MaxValidators)];
+                    for (int i = 0; i < votes.Length; i++)
+                        votes[i] = new BigInteger(r.ReadVarBytes());
+                    return new ValidatorsCountState
+                    {
+                        Votes = votes
+                    };
+                }
+            }
+
+            public byte[] ToByteArray()
```

Comment:
Perhaps this class should inherit from ISerializable, so we do this in a "standard" way.

Reply:
I have tried it. In that way, you will need an extra byte for serialization.